### DISCUSSION TITLE
Cannot upgrade from v2.3.4 to newest version

### DISCUSSION THREAD

**Discussion by roarnyg** [projectdiscovery/nuclei#1077] [Q&A] [answered]
I run the command 
_$ GO111MODULE=on go get -u -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei_
```
go: found github.com/projectdiscovery/nuclei/v2/cmd/nuclei in github.com/projectdiscovery/nuclei/v2 v2.5.2
go: github.com/golang/snappy upgrade => v0.0.4
go: github.com/golang/protobuf upgrade => v1.5.2
go: github.com/hashicorp/go-cleanhttp upgrade => v0.5.2
go: golang.org/x/net upgrade => v0.0.0-20210929193557-e81a3d93ecf6
go: github.com/mattn/go-isatty upgrade => v0.0.14
go: go.etcd.io/bbolt upgrade => v1.3.6
go: google.golang.org/protobuf upgrade => v1.27.1
...
```
Afterwards i type
_nuclei -v_
...and gets:
  _v2.3.4_

What have I tried so far?
I have tried removing everything from _/home/roar/go/pkg/mod/github.com/projectdiscovery/nuclei/_

_git clone https://github.com/projectdiscovery/nuclei.git; cd nuclei/cmd/nuclei/; go build; mv nuclei /usr/local/bin/; nuclei -version_

_go clean --modcache_

_echo $PATH | grep go_
`/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games:/usr/lib/go/../bin:/home/roar/go/bin:/home/roar/skript/sub404/:/home/roar/.go/bin:/home/roar/go/bin:/home/roar/skript`

I'm really stuck in this old nuclei version. It's still version 2.3.4

What other steps should I do to upgrade to newer version, for example v2.5.2?

**Comment by ehsandeep** [comment, marked as answer]
@roarnyg use `go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest` this is something we updated in "dev" version of the readme.

**Comment by roarnyg** [comment]
Thaks for helping me, but I get an error message: 
_go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest_                                                                                                                   
`package github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest: can only use path@version syntax with 'go get'`

I tinkered with the URL and tried:
_go get -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei/_                                                                                                                               
```
go: found github.com/projectdiscovery/nuclei/v2/cmd/nuclei in github.com/projectdiscovery/nuclei/v2 v2.5.2
github.com/projectdiscovery/fileutil
# github.com/projectdiscovery/fileutil
../pkg/mod/github.com/projectdiscovery/fileutil@v0.0.0-20210928100737-cab279c5d4b5/file.go:198:18: undefined: os.CreateTemp
```
and
_go get -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@v2.5.2_                                                                                                                          
```
go: downloading github.com/projectdiscovery/nuclei v2.5.2+incompatible
go: found github.com/projectdiscovery/nuclei/v2/cmd/nuclei in github.com/projectdiscovery/nuclei/v2 v2.5.2
github.com/projectdiscovery/fileutil
# github.com/projectdiscovery/fileutil
../pkg/mod/github.com/projectdiscovery/fileutil@v0.0.0-20210928100737-cab279c5d4b5/file.go:198:18: undefined: os.CreateTemp
```
_go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@v2.5.2_         
`package github.com/projectdiscovery/nuclei/v2/cmd/nuclei@v2.5.2: can only use path@version syntax with 'go get'`

When I type `nuclei -v` it still shows old nuclei version.

**Reply by geeknik** [reply]
What version of Go are you using?

**Reply by ehsandeep** [reply]
that error indicates you are running an older version of go, make sure you are on `go1.17`
